PayPalCheckout/src/main/java/com/paypal/android/checkout/paymentbutton/PaymentButton.kt

```diff
@@ -0,0 +1,281 @@
+package com.paypal.android.checkout.paymentbutton
+
+import android.content.Context
+import android.content.res.TypedArray
+import android.graphics.Canvas
+import android.os.Build
+import android.util.AttributeSet
+import android.util.TypedValue
+import android.view.Gravity
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.ImageView
+import android.widget.LinearLayout
+import android.widget.TextView
+import androidx.annotation.RequiresApi
+import androidx.core.content.ContextCompat
+import androidx.core.content.res.use
+import com.google.android.material.shape.CutCornerTreatment
+import com.google.android.material.shape.MaterialShapeDrawable
+import com.google.android.material.shape.RoundedCornerTreatment
+import com.google.android.material.shape.ShapeAppearanceModel
+import com.paypal.android.checkout.R
+
+@Suppress("TooManyFunctions")
+@RequiresApi(Build.VERSION_CODES.M)
+abstract class PaymentButton<C : PaymentButtonColor> @JvmOverloads constructor(
+    context: Context,
+    attributeSet: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : LinearLayout(context, attributeSet, defStyleAttr) {
+
+    /**
+     * Signals that the backing shape has changed and may require a full redraw.
+     */
+    private var shapeHasChanged = false
+
+    private var shapeAppearanceModel: ShapeAppearanceModel = ShapeAppearanceModel()
+        set(value) {
+            field = value
+            materialShapeDrawable = materialShapeDrawable.apply {
+                shapeAppearanceModel = field
+            }
+        }
+
+    private var materialShapeDrawable: MaterialShapeDrawable = MaterialShapeDrawable()
+        set(value) {
+            field = value
+            background = field
+        }
+
+    private val colorLuminance: PaymentButtonColorLuminance
+        get() = color.luminance
+
+    /**
+     * Updates the color of the Payment Button with the provided [PaymentButtonColor].
+     */
+    abstract var color: C
+
+    /**
+     * Provides the current value of the prefix text which is displayed before the button's wordmark.
```

Comment:
Nitpick - can we remove "Proivdes the current" on each of theses docs and just have "The prefix text..." etc for these strings? The extra text seems unnecessary.

Reply:
Sounds good to me